File: ruby/difference-of-squares/difference_of_squares.rb
class BookKeeping
  VERSION = 3
end

class Squares
  attr_reader :number

  def initialize(number)
    @number = number
  end

  def square_of_sum
    (1..number).reduce(:+) ** 2
  end

  def sum_of_squares
    (1..number).map { |n| n ** 2 }.reduce(:+)
  end

  def difference
    return 0 if number == 0

    square_of_sum - sum_of_squares
  end
end

File: typescript/difference-of-squares/difference-of-squares.ts
class Squares {
  input: number[];
  squareOfSum: number;
  sumOfSquares: number;
  difference: number;

  constructor(input: number) {
    this.input = [...Array(input + 1).keys()];
    this.squareOfSum = this.input.reduce((acc, num) => acc + num, 0) ** 2;
    this.sumOfSquares = this.input.reduce((acc, num) => acc + num ** 2, 0);
    this.difference = this.squareOfSum - this.sumOfSquares;
  }
}

export default Squares;
